magda-web-client/src/helpers/VocabularyApis.ts

```diff
@@ -0,0 +1,84 @@
+import { config } from "../config";
+import fetch from "isomorphic-fetch";
+import uniq from "lodash/uniq";
+import flatMap from "lodash/flatMap";
+const apiEndpoints = config.vocabularyApiEndpoints;
+
+async function queryEndpoint(
+    apiEndpoint: string,
+    str: string
+): Promise<string[]> {
+    if (!apiEndpoint) {
+        throw new Error(
+            "Failed to contact Vocabulary API: API endpoint cannot be empty!"
+        );
+    }
+
+    const requestUrl =
+        `${apiEndpoint}?labelcontains=` + encodeURIComponent(str);
+
+    const data = await fetch(requestUrl).then(response => {
+        if (response.status === 200) {
+            return response.json();
+        }
+        throw new Error(response.statusText);
+    });
+
+    if (!data || !data.result || !Array.isArray(data.result.items)) {
+        throw new Error(
+            "Failed to contact Vocabulary API: Invalid API response!"
+        );
+    }
+
+    const keywords: string[] = [];
+
+    data.result.items.forEach(item => {
+        let prefLabels: {
+            _value: string;
+            _lang: string;
+        }[] = [];
+
+        if (item.broader && item.broader.prefLabel) {
+            if (!Array.isArray(item.broader.prefLabel)) {
```

Comment:
I think it'd work a bit better if we included the other labels too, not just the pref labels?

Reply:
As discussed, we leave this unchanged until we found other fields that want to be included